x-pack/plugins/uptime_monitoring/public/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import ReactDOM from 'react-dom';
+import { UMFrameworkAdapter } from '../../lib';
+
+export class UMKibanaFrameworkAdapter implements UMFrameworkAdapter {
+  private uiRoutes: any;
+
+  constructor(uiRoutes: any) {
+    this.uiRoutes = uiRoutes;
+  }
+
+  public render = (component: React.ReactElement<any>) => {
+    this.register(component);
+  };
+
+  private register = (rootComponent: React.ReactElement<any>) => {
```

Comment:
The mounting you do here is good, but it does not take into consideration the needed unmounting. Otherwise angular will keep this in memory and navigating away and back will result in multiple allocations. To avoid this you will need something like the code we have in BeatsCM:
```
private manageAngularLifecycle($scope: any, $route: any, elem: any) {
    const lastRoute = $route.current;
    const deregister = $scope.$on('$locationChangeSuccess', () => {
      const currentRoute = $route.current;
      // if templates are the same we are on the same route
      if (lastRoute.$$route.template === currentRoute.$$route.template) {
        // this prevents angular from destroying scope
        $route.current = lastRoute;
      }
    });
    $scope.$on('$destroy', () => {
      if (deregister) {
        deregister();
      }
      // manually unmount component when scope is destroyed
      if (elem) {
        ReactDOM.unmountComponentAtNode(elem);
      }
    });
  }
```

Reply:
Does https://github.com/elastic/kibana/pull/25480/commits/42a087e19f4a78b9fbe7858ec1976225d86aef9a look like a step in the right direction?